### PR TITLE
chore(dev-deps): Add rubocop-performance

### PR DESCRIPTION
**Pull request by kaylareopelle**
This PR adds the [`rubocop-performance` gem](https://rubygems.org/search?query=rubocop-performance) to all contrib gems. It also updates the code to fulfill the linter's recommendations.

### CODE REVIEW

File: instrumentation/grape/lib/opentelemetry/instrumentation/grape/event_handler.rb [hunk 1]
```diff
@@ -79,7 +79,7 @@ def handle_payload_exception(span, exception)
             # Only record exceptions if they were not raised (i.e. do not have a status code in Grape)
             # or do not have a 5xx status code. These exceptions are recorded by Rack.
             # See instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb#L155
-            return unless exception.respond_to?('status') && ::Rack::Utils.status_code(exception.status) < 500
+            return unless exception.respond_to?(:status) && ::Rack::Utils.status_code(exception.status) < 500
```
Comment (arielvalentin) [inline instrumentation/grape/lib/opentelemetry/instrumentation/grape/event_handler.rb:82]: Odd. this should not make a difference if we are using frozen string literals right?

**Review by arielvalentin — APPROVED** [review]

File: instrumentation/grape/lib/opentelemetry/instrumentation/grape/event_handler.rb (hunk 1 above)
Comment (kaylareopelle) [reply instrumentation/grape/lib/opentelemetry/instrumentation/grape/event_handler.rb:82]: Good question! I'll do a bit of digging.

Comment (kaylareopelle) [reply instrumentation/grape/lib/opentelemetry/instrumentation/grape/event_handler.rb:82]: So, I don't understand why, yet, but I can confirm with `Benchmark.ips` and `Benchmark.ipsa` that it is faster even when using frozen string literals. [Here's a gist](https://gist.github.com/kaylareopelle/e8e1f04cb1c9fb399b38e27f82d390da) with the code I ran using Ruby 3.2.2 using ipsa. 

I ran it a few times and strings were always between 1.5 - 1.8 times slower.

Here's the original PR for that cop: https://github.com/rubocop/rubocop-performance/pull/276 

It sounds like it's from Polished Ruby Programming, which I have at home, so I'll see if I can learn more there.

Comment (arielvalentin) [reply instrumentation/grape/lib/opentelemetry/instrumentation/grape/event_handler.rb:82]: Amazing. String Literals always return the same object id whereas symbols do not. 

This surprises me so much!!!

Thank goodness for these linters.

**Pull request merged**